File: Match.java
import java.util.Random;

public class Match {
    private Team team1;
    private Team team2;

    public Match(Team team1, Team team2) {
        this.team1 = team1;
        this.team2 = team2;
    }

    public void simulate() {
        Random rand = new Random();
        int team1Goals = rand.nextInt(5); // 0-4 random goals
        int team2Goals = rand.nextInt(5);

        System.out.println("Match Result: " + team1.getName() + " " + team1Goals + " - " + team2Goals + " " + team2.getName());

        team1.updateStats(team1Goals, team2Goals);
        team2.updateStats(team2Goals, team1Goals);
    }
}

File: match.js
class Match {
  constructor(team1, team2) {
    this.team1 = team1;
    this.team2 = team2;
    this.team1Goals = Math.floor(Math.random() * 5); // Rastgele gol sayısı
    this.team2Goals = Math.floor(Math.random() * 5);
  }

  simulate() {
    this.team1.updateStats(this.team1Goals, this.team2Goals);
    this.team2.updateStats(this.team2Goals, this.team1Goals);
  }
}

class MatchQueue {
  constructor() {
    this.queue = [];
  }

  // Maçı sıraya ekle
  addMatch(match) {
    this.queue.push(match);
  }

  // Sıradaki maçı al
  getNextMatch() {
    return this.queue.shift(); // Kuyruğun başından çıkar
  }
}

class MatchHistory {
  constructor() {
    this.history = [];
  }

  // Maçı geçmişe ekle
  addMatch(match) {
    this.history.push(match);
  }

  // Son maçı getir
  getLastMatch() {
    return this.history.length > 0
      ? this.history[this.history.length - 1]
      : null;
  }
}

module.exports = Match;
